ArduPlane/sensors.cpp

```diff
@@ -65,6 +65,10 @@ void Plane::accel_cal_update() {
  */
 void Plane::read_airspeed(void)
 {
+    static int airspeed_failed_count=0;
```

Comment:
@WickedShell its not only about the time but the fact that we've built up a number of failures, it just happens that they are 3 seconds worth of failures.  If there is a way to know the rate that it's getting called?  If so I'd rather keep the count, but have the check of the counter against 3 seconds worth of failures if that makes sense.  Note this will be moving from sensors.cpp as per the dev call.

Reply:
static variables are not allowed. If The Plane class was ever created twice it would called problems. We don't support that now, but it may be an option in the future.. so best to do it right just in case. Also, never use counters. Always use time based events.

Reply:
Remove the static's and put the counters in the airspeed state instead.  I absolutely need the counters, I need to know when we have accumulated enough errors.  It needs to build up a number of errors over a short duration.  I'll add a clearing feature so that every minute or two it clears the counters, to avoid a build up over a long period of time.

Reply:
again, if we decide to change this to 10 or 50Hz or 400Hz then the algorithm is hozed and it's difficult to track down. Try a low-pass filter on a probability like the is_flying flag
https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/is_flying.cpp#L130

Reply:
I like that method using the low pass filter, so I then just need to set the desired percentage for the failure and recovery.  Thank you @magicrub that is a much better way of doing it.